### PR TITLE
#4631 [Manager] Refactor the Dialog and DeleteDialog components

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4631 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by eechih** [comment]
retry manager

File: ohara-manager/client/src/components/Autofill/AutofillEditor.js [hunk 1]
```diff
@@ -118,11 +118,11 @@ const AutofillEditor = props => {
           <Dialog
             confirmDisabled={submitting || pristine || invalid}
             confirmText="Save"
-            handleClose={() => {
+            onClose={() => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Autofill/AutofillEditor.js:121]: Any reason for renaming this?

File: ohara-manager/client/src/components/common/Dialog/Dialog.js [hunk 2]
```diff
@@ -59,73 +60,81 @@ const Progress = styled(CircularProgress)`
...
+  testId,
+  title,
+}) => (
+  <MuiDialog
+    data-testid="dialog-container"
+    fullWidth
+    maxWidth={maxWidth}
+    onClose={onClose}
+    open={open}
+    PaperComponent={DrabblePaper}
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Dialog/Dialog.js:83]: @eechih We can use an eslint rule to enforce the prop sorting... manually doing this is exhausting 😅

Comment (konekoya) [reply ohara-manager/client/src/components/common/Dialog/Dialog.js:83]: Never use this before, but should fit our need?
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md

File: ohara-manager/client/src/components/Autofill/AutofillEditor.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/Autofill/AutofillEditor.js:121]: Just follow MuiTable props :)

File: ohara-manager/client/src/components/common/Dialog/Dialog.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/Dialog/Dialog.js:83]: > We can use an eslint rule to enforce the prop sorting... manually doing this is exhausting

same feeling 😅

Comment (konekoya) [reply ohara-manager/client/src/components/common/Dialog/Dialog.js:83]: Okay, i will open an issue to add this rule which will enforce the prop sorting :)

File: ohara-manager/client/src/components/Autofill/AutofillEditor.js (hunk 1 above)
Comment (konekoya) [reply ohara-manager/client/src/components/Autofill/AutofillEditor.js:121]: Hmm, right. Didn't think about this before... just thought it's a wrapped component. But using Mui's prop names seem more intuitive and clear. Nice call 👍

File: ohara-manager/client/src/components/common/Dialog/Dialog.js (hunk 2 above)
Comment (konekoya) [reply ohara-manager/client/src/components/common/Dialog/Dialog.js:83]: #4638

**Pull request merged**